lib/ZwaveClient.ts

```diff
@@ -3355,7 +3359,10 @@ class ZwaveClient extends TypedEventEmitter<ZwaveClientEventCallbacks> {
 	private _onNodeNotification(
 		zwaveNode: ZWaveNode,
 		ccId: CommandClasses,
-		args: Record<string, unknown>
+		args:
+			| ZWaveNotificationCallbackArgs_NotificationCC
+			| ZWaveNotificationCallbackArgs_EntryControlCC
+			| ZWaveNotificationCallbackArgs_MultilevelSwitchCC
 	) {
```

Comment:
I think you'd better use the official callback type for this. That way the dependencies between the `ccId`  and the `args`  parameters are preserved:
https://github.com/zwave-js/node-zwave-js/blob/3b5c91a19e09132900dfd7000fd53f8409f2cb0f/packages/zwave-js/src/lib/node/_Types.ts#L87-L93

```ts
private _onNodeNotification: ZWaveNotificationCallback = (zwaveNode, ccId, args) => {
  // ...
}
```

Leaves less room for type errors on your side, or forgetting to update the signature when the types change.

Reply:
are you sure the `this` is preserved correctly inside the function declared in that way?

Reply:
done